tests/typing/aggregate.py

```diff
@@ -8,29 +9,36 @@ class Cat(Animal): ...
 
 
 # Test 1: to check Aggregate provider
-provider1: providers.Aggregate[str] = providers.Aggregate(
+provider1 = providers.Aggregate(
     a=providers.Object("str1"),
     b=providers.Object("str2"),
 )
-provider_a_1: providers.Provider[str] = provider1.a
+provider_a_1 = provider1.a
 provider_b_1: providers.Provider[str] = provider1.b
-val1: str = provider1("a")
+val1 = provider1("a")
+assert_type(provider1, providers.Aggregate[str])
+assert_type(provider_a_1, providers.Provider[str])
+assert_type(provider_b_1, providers.Provider[str])
+assert_type(val1, str)
 
-provider1_set_non_string_keys: providers.Aggregate[str] = providers.Aggregate()
+provider1_set_non_string_keys = providers.Aggregate[str]()
 provider1_set_non_string_keys.set_providers({Cat: providers.Object("str")})
-provider_set_non_string_1: providers.Provider[str] = (
-    provider1_set_non_string_keys.providers[Cat]
-)
+provider_set_non_string_1 = provider1_set_non_string_keys.providers[Cat]
+assert_type(provider_set_non_string_1, providers.Provider[str])
+
 
-provider1_new_non_string_keys: providers.Aggregate[str] = providers.Aggregate(
+# TODO: Change providers.Aggregate to accept Mapping? Then remove explicit typing here
```

Comment:
Current stub:

```python
class Aggregate(Provider[T]):
    def __init__(
        self,
        provider_dict: Optional[_Dict[Any, Provider[T]]] = None,
        **provider_kwargs: Provider[T],
    ): ...
```

I think we should accept `collections.abc.Mapping` here

```python
class Aggregate(Provider[T]):
    def __init__(
        self,
        provider_dict: Optional[Mapping[Any, Provider[T]]] = None,
        **provider_kwargs: Provider[T],
    ): ...
```

Because `Dict` is invariant in both key and value, so `providers.Aggregate` cannot infer its own typevar without explicit typing

```python
provider1_new_non_string_keys = providers.Aggregate(
    {Cat: providers.Object("str")},
)  # This raises mypy error
```